pkgs/development/cuda-modules/_cuda/db/ingest_releases_file.nix

```diff
@@ -0,0 +1,90 @@
+# NOTE: belongs in cudaLib
+# NOTE: cf. tensorrt/releases.nix for a sample input
+# NOTE: included in db/default.nix
+let
+  inherit (import ./nixpkgs_paths.nix) libPath;
+  inherit (import ./columnar.nix)
+    unit
+    ;
+in
+
+{
+  lib ? import libPath,
+  baseUrlAttr,
+}:
+
+arg:
+
+let
+  attrs = if builtins.isPath arg then import arg else arg;
+in
+
+assert lib.isAttrs attrs;
+assert lib.all (child: child ? releases) (lib.attrValues attrs);
+
+{ config, ... }:
+
+rec {
```

Comment:
It was my understanding that using the `rec` keyword instead of using `config` to create recursive relationships breaks overriding, since `rec` will just present a snapshot of the value, rather than using the value produced by the fixed-point, like going through `config` will.
Will that be a problem for this module? If the configuration is meant to be supplied in full, probably not; but if users are able to modify one or two values and then expect dependent values to change, they might be surprised that they do not.